src/transactions/conversion-check.guard.ts

```diff
@@ -1,29 +1,46 @@
-import {Injectable, CanActivate, ExecutionContext, UnauthorizedException, BadRequestException} from '@nestjs/common';
+import {
+	Injectable,
+	CanActivate,
+	ExecutionContext,
+	UnauthorizedException,
+	BadRequestException,
+	ForbiddenException
+} from '@nestjs/common';
+import {getRepository} from 'typeorm';
 import {SignedInBot} from 'types/bot';
+import {Currency} from 'src/currencies/currency.entity';
 import {Transaction} from './transaction.entity';
 
 @Injectable()
 export class ConversionCheckGuard implements CanActivate {
 	async canActivate(context: ExecutionContext): Promise<boolean> {
-		const req: {body?: Transaction; user?: SignedInBot} = context.switchToHttp().getRequest();
+		const req: {
+			body?: Transaction;
+			user?: SignedInBot;
+		} = context.switchToHttp().getRequest();
 		const signedInBot = req.user;
 		const {body} = req;
+		const currencies = getRepository(Currency);
 
 		if (!signedInBot) {
 			throw new UnauthorizedException();
 		}
 
 		if (body) {
+			const toCurrency = await currencies.findOne(body.toId);
 			if (body.toId === signedInBot.currency.id) {
 				throw new BadRequestException(
 					`You can not convert ${signedInBot.currency.id} to ${body.toId} because they are the same`
 				);
+			} else if (!toCurrency) {
+				throw new BadRequestException(`Currency ${body.toId} does not exist`);
+			} else if ((body.amount * signedInBot.currency.value) / toCurrency.value >= toCurrency.reserve) {
```

Comment:
This is assuming that the provided body actually has an `amount` property.

Reply:
https://github.com/Discoin/api-v3/commit/c4dc2edc7d102cdbf229deb18dbcfa6fdc645690